packages/block-editor/src/components/media-upload-modal/index.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * WordPress dependencies
+ */
+import { withFilters } from '@wordpress/components';
+
+/**
+ * This is a placeholder for the media upload modal component necessary to make it possible to provide
+ * an integration with the core blocks that handle media files. By default it renders nothing but
+ * it provides a way to have it overridden with the `editor.MediaUploadModal` filter.
+ *
+ * @return {Component} The component to be rendered.
+ */
+const MediaUploadModal = () => null;
```

Comment:
Very minor as well, but I wonder if it could be called something like `DataViewsMediaModal` to distinguish it a bit more from the other component.